pkg/nmi/server/endpointpolicy.go

```diff
@@ -79,11 +89,12 @@ func getEndpointByIP(ip string) (*v1.HNSEndpoint, error) {
 		}
 	}
 
-	return nil, fmt.Errorf("No endpoint found for IP address - %s", ip)
+	klog.Warningf("No endpoint found for Pod IP - %s.", ip)
+	return nil, nil
```

Comment:
>nil [](start = 13, length = 3)

This should throw error, so that the call can handle it (decide if this is fine for its case).

Reply:
In fact, I would like to distinguish three cases in this function:
1. Successful
2. Calling agent failed error
3. Did not find the endpoint

So for case 3, I log it as a warning and return nil for both endpoint and error. In this case, the caller can distinguish. I think return error for case 3 is also fine, then the caller needs to distinguish the error type, which is a little more complex than the current implementation. How do you think?